File: frontend/src/types/telegram_idea.ts
/**
 * 텔레그램 아이디어 타입 정의
 */

export type IdeaSourceType = 'my' | 'others'

export interface TelegramIdea {
  id: string
  created_at: string

  // 채널 정보
  channel_id: number
  channel_name: string
  source_type: IdeaSourceType

  // 메시지 정보
  message_id: number
  message_text: string
  original_date: string

  // 포워딩 정보
  is_forwarded: boolean
  forward_from_name: string | null

  // 종목 정보
  stock_code: string | null
  stock_name: string | null

  // 분석 데이터
  sentiment: 'POSITIVE' | 'NEGATIVE' | 'NEUTRAL' | null
  sentiment_score: number | null

  // 메타데이터
  raw_hashtags: string[]
}

export interface TelegramIdeaListResponse {
  items: TelegramIdea[]
  total: number
  offset: number
  limit: number
}

export interface StockMentionStats {
  stock_code: string
  stock_name: string
  mention_count: number
  latest_date: string
  sources: IdeaSourceType[]
}

export interface StockStatsResponse {
  stocks: StockMentionStats[]
  total_count: number
}

export interface AuthorStockInfo {
  stock_code: string
  stock_name: string
  count: number
}

export interface AuthorStats {
  name: string
  idea_count: number
  top_stocks: AuthorStockInfo[]
  latest_idea_date: string
}

export interface AuthorStatsResponse {
  authors: AuthorStats[]
  total_count: number
}

export interface TelegramIdeaCollectResult {
  channel_name: string
  messages_collected: number
  ideas_created: number
  errors: string[]
}

export interface TelegramIdeaCollectResponse {
  results: TelegramIdeaCollectResult[]
  total_messages: number
  total_ideas: number
}

File: backend/schemas/telegram_idea.py
"""텔레그램 아이디어 스키마."""
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class TelegramIdeaResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    created_at: datetime
    channel_id: int
    channel_name: str
    source_type: str
    message_id: int
    message_text: str
    original_date: datetime
    is_forwarded: bool
    forward_from_name: Optional[str] = None
    stock_code: Optional[str] = None
    stock_name: Optional[str] = None
    sentiment: Optional[str] = None
    sentiment_score: Optional[float] = None


class TelegramIdeaListResponse(BaseModel):
    items: list[TelegramIdeaResponse]
    total: int
    offset: int
    limit: int


class StockMentionStats(BaseModel):
    stock_code: str
    stock_name: Optional[str] = None
    mention_count: int
    latest_date: Optional[datetime] = None
    sources: Optional[list[str]] = None


class StockStatsResponse(BaseModel):
    stocks: list[StockMentionStats]
    total_count: int


class TopStockItem(BaseModel):
    stock_code: str
    stock_name: Optional[str] = None
    count: int


class AuthorStats(BaseModel):
    name: str
    idea_count: int
    top_stocks: Optional[list[TopStockItem]] = None
    latest_idea_date: Optional[datetime] = None


class AuthorStatsResponse(BaseModel):
    authors: list[AuthorStats]
    total_count: int


class TraderPick(BaseModel):
    stock_code: str
    stock_name: str
    return_pct: float
    mention_date: str


class TraderRankingItem(BaseModel):
    rank: int
    name: str
    idea_count: int
    avg_return_pct: float
    win_rate: float
    total_return_pct: float
    best_pick: Optional[TraderPick] = None
    worst_pick: Optional[TraderPick] = None
    picks: list[TraderPick] = []


class TraderRankingResponse(BaseModel):
    traders: list[TraderRankingItem]
    total_traders: int
    analysis_period_days: int
    min_mentions: int


class CollectResult(BaseModel):
    channel_name: str
    messages_collected: int
    ideas_created: int
    errors: list[str]


class CollectResponse(BaseModel):
    results: list[CollectResult]
    total_messages: int
    total_ideas: int
